### PR TITLE
ISAICP-3727: Style featured cue

### PR DESCRIPTION
**Pull request by tom-waat**

### CODE REVIEW

File: web/modules/custom/joinup_community_content/js/pinned_content.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
    *   The HTML for the pinned content visual cue.
    */
   Drupal.theme.pinnedContentCue = function () {
-    return '<div class="listing__pin"><span class="icon icon--pin"></span></div>';
+    return '<div class="listing__corner listing__corner--pin"><span class="icon icon--pin"></span></div>';
```
Comment (kalinchernev) [inline web/modules/custom/joinup_community_content/js/pinned_content.js]: I like the idea of renaming the `listing__pin` to `listing__corner`.
Please remove `listing__corner--pin`, it's not necessary.

Comment (kalinchernev) [reply web/modules/custom/joinup_community_content/js/pinned_content.js]: On another, slightly related topic - when I both feature and pin a tile, only the pin shows and there is no featured tile behavior.

File: web/modules/custom/joinup_community_content/js/pinned_content.js [hunk 2]
```diff
@@ -42,9 +42,10 @@
           return;
         }
 
+        $this.addClass('listing__card--corner');
         // It is needed for listing title padding if image doesn't exist.
         if (!$this.find('.listing__image').length) {
-          $this.addClass('listing__card--title-pin');
+          $this.addClass('listing__card--corner-title');
```
Comment (kalinchernev) [inline web/modules/custom/joinup_community_content/js/pinned_content.js]: I tried the scenario where there is no image, here's how the helper fixes this:
tile_with_helper_class

And this is how it looks without a helper:
tile_with_no_helper_class

Are you sure this helper is actually helping for better visualization?

File: web/profiles/joinup/js/site_wide_featured.js [hunk 3]
```diff
@@ -6,6 +6,16 @@
 (function ($, Drupal) {
 
   /**
+   * Theme function that renders a visual cue for featured content.
+   *
+   * @returns {string}
+   *   The HTML for the featured content visual cue.
+   */
+  Drupal.theme.featuredContentCue = function () {
+    return '<div class="listing__corner listing__corner--featured"><span class="icon icon--star"></span></div>';
```
Comment (kalinchernev) [inline web/profiles/joinup/js/site_wide_featured.js]: `listing__corner--featured` seems be unnecessary - I couldn't find any reference usage for it, similarly to the `listing__corner--pin`.

File: web/profiles/joinup/js/site_wide_featured.js [hunk 4]
```diff
@@ -24,7 +34,16 @@
...
       }
 
-      $(context).find('[data-drupal-featured]').once('featured').addClass('is-featured');
+      $(context).find('[data-drupal-featured]').once('featured').each(function () {
+        var $this = $(this);
+        $this.addClass('is-featured');
+        $this.addClass('listing__card--corner');
+
+        if (!$this.find('.listing__image').length) {
+          $this.addClass('listing__card--corner-title');
```
Comment (kalinchernev) [inline web/profiles/joinup/js/site_wide_featured.js]: Please check if this is still needed.

File: web/themes/joinup/prototype/sass/components/_icon.sass [hunk 5]
```diff
@@ -155,6 +155,9 @@ $waat-icons-font-path: '../fonts/waat-icons/waat-icons'
 .icon--github:before
   content: '\e8d6'
 
+.icon--star:before
```
Comment (kalinchernev) [inline web/themes/joinup/prototype/sass/components/_icon.sass]: Missing inclusion on the top of the file which will help on prototype demo.

File: web/themes/joinup/prototype/sass/components/_listing.sass [hunk 6]
```diff
@@ -171,7 +171,7 @@
 .listing__title--link a
   color: $color-orange
 
-.listing__card--title-pin
+.listing__card--corner-title
```
Comment (kalinchernev) [inline web/themes/joinup/prototype/sass/components/_listing.sass]: I believe that could have been necessary when the contextual navigation menu has been on the right and the icon has been pushing the menu. But now the contextual navigation menu is on the left when the tile is either pinned or featured, so I don't think this is an actually useful tweak any more.

File: web/profiles/joinup/js/site_wide_featured.js [hunk 7]
```diff
@@ -24,7 +34,16 @@
         return;
       }
 
-      $(context).find('[data-drupal-featured]').once('featured').addClass('is-featured');
+      $(context).find('[data-drupal-featured]').once('featured').each(function () {
+        var $this = $(this);
+        $this.addClass('is-featured');
+        $this.addClass('listing__card--corner');
```
Comment (kalinchernev) [inline web/profiles/joinup/js/site_wide_featured.js]: Those 2 can be either changed or added in 1 go to save 1 DOM manipulation :)

File: web/modules/custom/joinup_community_content/js/pinned_content.js (hunk 1 above)
Comment (luke-waat) [reply web/modules/custom/joinup_community_content/js/pinned_content.js]: Class has been removed.
Regarding this behaviour you need to contact Marco or Francesco, but I am quite sure that this is defined and implemented correctly. Tom replied to my text message that there is no possibility to have featured and pinned tiles at once.

File: web/modules/custom/joinup_community_content/js/pinned_content.js (hunk 2 above)
Comment (luke-waat) [reply web/modules/custom/joinup_community_content/js/pinned_content.js]: In my opinion this helper is useful. Your title fits perfectly into the tiles, if you will add few more letters in first line the title will overlap the icon.

File: web/profiles/joinup/js/site_wide_featured.js (hunk 3 above)
Comment (luke-waat) [reply web/profiles/joinup/js/site_wide_featured.js]: Done.

File: web/profiles/joinup/js/site_wide_featured.js (hunk 4 above)
Comment (luke-waat) [reply web/profiles/joinup/js/site_wide_featured.js]: Let's leave it as it is.

File: web/themes/joinup/prototype/sass/components/_icon.sass (hunk 5 above)
Comment (luke-waat) [reply web/themes/joinup/prototype/sass/components/_icon.sass]: New icons have been added to the styleguide.

File: web/themes/joinup/prototype/sass/components/_listing.sass (hunk 6 above)
Comment (luke-waat) [reply web/themes/joinup/prototype/sass/components/_listing.sass]: Let's leave it as it is. This helper prevent overlapping of the pinned / featured icon and the title.

File: web/profiles/joinup/js/site_wide_featured.js (hunk 7 above)
Comment (luke-waat) [reply web/profiles/joinup/js/site_wide_featured.js]: Done.

**Pull request merged**